### ISSUE TITLE
setVideoPath Crashing

### ISSUE THREAD

**Issue by Rensodiaz** [rtoshiro/FullscreenVideoView#2]
Hi,
Every time I try to set a video from a local path the application crash with this log:

java.lang.IllegalStateException: FullscreenVideoView Invalid State: PREPARING

It works great when I set an Uri.

**Comment by Rensodiaz** [comment]
I found the problem, ty.

**Comment by kcochibili** [comment]
what was your solution?
i am having the same problem. In my case, the device freezes. each time I attempt to play a video from file system.
so how did you solve it?

**Comment by Rensodiaz** [comment]
I had a playlist of videos in the SD and I was looping through it without waiting, so it call this method multiple times. 
    `java
        videoLayout.setVideoPath(mPathVideo);
`
there is a couple of listeners that you can use to handle this situation that this library give you, example:

``````
```java
     videoLayout.setOnPreparedListener(new MediaPlayer.OnPreparedListener() {
         @Override
         public void onPrepared(MediaPlayer mp) {
             Log.i(tag, "Prepare");
             Log.i(tag, "duration: " + videoLayout.getDuration());
        }
    });
```
``````

Let me know if it helps.

**Comment by kcochibili** [comment]
It turned out my problem and your problem had the same roots.

The problem turned out to be that I did not call `VideoView.stopPlayback()` in the previous activity where I had a handle on this video file (The video was still playing when I launched a new activity that attempted to play the same video Using `FullscreenVideoLayout`).

So I was able to stop the freezing by first calling  `VideoView.stopPlayback()` before launching a new activity.

Thank you for the helpful diagnostic suggestion.that helped solved my problem.
